src/NuGet.Core/NuGet.Packaging/LocalPackageArchiveDownloader.cs

```diff
@@ -144,23 +144,20 @@ public void Dispose()
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            // This value is copied from System.IO.Stream.
+            // It is the largest multiple of 4096 that is smaller than the large object heap (LOH)
+            // threshold of 85K and offers better copy performance than a 4K buffer.
+            const int bufferSize = 81920;
+
             using (var destination = new FileStream(
                 destinationFilePath,
                 FileMode.Create,
-                FileAccess.ReadWrite,
-                FileShare.ReadWrite | FileShare.Delete,
```

Comment:
@jainaashish, I'll make this change at a later time.  It's not perf-related.  This method copies a .nupkg and read access and read/delete sharing is unnecesssary.  This code dates from before plugin changes, and at that time the same stream was used for writing, reading and deleting.  In a post-plugin world, that code has been split up.